.github/actions/java-maven-setup/action.yml

```
@@ -0,0 +1,29 @@
+name: 'Java + Maven configuration'
+description: 'Java and Maven version setup'
+inputs:
+  java-version:
+    description: "the java version"
+    default: 11
+    required: false
+  maven-version:
+    description: "the maven version"
+    default: "3.8.1"
+    required: false
+  cache-key-prefix:
+    description: "the cache key"
+    required: false
```

Let's add a default value to avoid `-m2` as key (${{ inputs.cache-key-prefix }}-m2)
```suggestion
    default: "cache-key"
    required: false
```